File: scripts/py-compile-suite/inputs/23_find_where.m
% Find indices
x = [1, -2, 3, -4, 5, -6, 7];
positive_idx = find(x > 0);
first_neg = find(x < 0, 1, 'first');
last_neg = find(x < 0, 1, 'last');
positive_vals = x(positive_idx);
count_pos = length(positive_idx);

File: scripts/py-compile-suite/outputs/23_find_where.py
import numpy as np

# Find indices
x = np.array([1, -2, 3, -4, 5, -6, 7])
positive_idx = np.flatnonzero(x > 0)
first_neg = np.flatnonzero(x < 0)[0]
last_neg = np.flatnonzero(x < 0)[-1]
positive_vals = x[positive_idx - 1]
count_pos = np.max(positive_idx.shape)
